torchao/dtypes/nf4tensor.py

```diff
@@ -698,7 +699,7 @@ def quantize_tensor_nearest(
         value = value.unsqueeze(-1)  # (numel, 1)
         # Compare the value tensor with the nf4 tensor element-wise
         diff = (value - nf4).abs()
-        closest_nf4 = diff.min(dim=-1).indices
+        closest_nf4 = diff.min(dim=-1).indices.to(dtype=torch.uint8)
```

Comment:
Regarding, `diff.min(dim=-1).indices.to(torch.uint8)`: if `diff.argmin(dim = -1, dtype = torch.uint8)` [is supported](https://github.com/pytorch/pytorch/issues/32570), this might get neater